### DISCUSSION TITLE
handleViewStateChange	@ deckgl.ts:155124 Warning: Maximum update depth exceeded.

### DISCUSSION THREAD

**Discussion by damianjchocron** [visgl/deck.gl#7835] [Q&A]
I have to use a custom handleViewChange but when I move using fly() (a lot of change of state per second) I have this error.

Looks like Reac detect this amount of change of state as a problem, but It's the only way to move and have control of the currentViewState

`
      <DeckGL
          viewState={currentViewState}
          controller={CUSTOM_CONTROLLER}
         onViewStateChange={({viewState, oldViewState}) =>
        handleViewStateChange(viewState, oldViewState)
      }
      onClick={() => HandleOnClick()} // <- When clickout sample, hide tooptip}
    >
`

`
const handleViewStateChange = useCallback((viewState, oldViewState) => {
      setCurrentViewState({
        ...viewState,
        transitionDuration: INITIAL_VIEW_STATE.transitionDuration,
      }); //<- add transitionDuration is a workAround to keep this value
    }
  },[currentViewState]);
`

Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
(anonymous) | @ | instrument.ts:124

  | overrideMethod | @ | react_devtools_backend.js:2655
  | printWarning | @ | react-dom.development.js:86
  | error | @ | react-dom.development.js:60
  | checkForNestedUpdates | @ | react-dom.development.js:27300
  | scheduleUpdateOnFiber | @ | react-dom.development.js:25475
  | dispatchSetState | @ | react-dom.development.js:17527
  | (anonymous) | @ | Map.js:809
  | onViewStateChange | @ | DeckGLCustom.js:60
  | handleViewStateChange | @ | deckgl.ts:155
  | _onViewStateChange | @ | deck.ts:1039
  | _onViewStateChange | @ | view-manager.ts:283
  | _onTransition | @ | controller.ts:393
  | (anonymous) | @ | transition-manager.ts:232
  | update | @ | transition.ts:96
  | updateTransition | @ | transition-manager.ts:120
  | updateTransition | @ | controller.ts:349
  | updateViewStates | @ | view-manager.ts:126
  | _onRenderFrame | @ | deck.ts:1031

**Comment by Pessimistress** [comment]
You should not overwrite anything in the `viewState` unless you know what you are doing.

**Reply by damianjchocron** [reply]
I'm using Deck for a long time in his way because I really need to have control of the `viewState` all the time.
Is this _Warning: Maximum update depth exceeded_ a bug from my side or from the library?

**Reply by Pessimistress** [reply]
You are not supposed to overwrite `transitionDuration` blindly because it is managed by the controller. If you do believe that there is a bug in the library, create a Code Sandbox that reproduces your issue. I cannot help you with what little code you shared.

**Reply by damianjchocron** [reply]
I made the overwrite [of this because](https://github.com/visgl/deck.gl/issues/7158#issuecomment-1329722960):
I'll create a Code Sandbox :)
Thanks!
